skglm/estimators.py

```diff
@@ -22,7 +22,7 @@
 from skglm.solvers import AndersonCD, MultiTaskBCD
 from skglm.datafits import Cox, Quadratic, Logistic, QuadraticSVC, QuadraticMultiTask
 from skglm.penalties import (L1, WeightedL1, L1_plus_L2, L2,
-                             MCPenalty, IndicatorBox, L2_1)
+                             MCPenalty, WeightedMCPenalty, IndicatorBox, L2_1)
```

Comment:
@Badr-MOUFAD @PABannier : do you think we should implement a new penalty, or simply support weights in the existing `MCPenalty` ?

Reply:
I don't have a strong opinion but I would stick with the current design and have a new `WeightedMCPenalty`

Reply:
I did a try with adding `weights` in the `MCPenalty`, and it starts to be ugly and a pain to add 'weights=np.ones(n_features)' everywhere.
Having only one `MCPRegression` with a `weights` option, and having two penalties `MCPenalty` and `WeightedMCPenalty` seems to be the cleanest option

Reply:
Agreed with @QB3 !